gladier/client.py

```diff
@@ -59,53 +55,35 @@ class GladierBaseClient(object):
+
         self._tools = None
-        self.public_config = self._load_public_config()
-        self.private_config = self._load_private_config()
-        self.authorizers = authorizers or dict()
-        self.auto_login = auto_login
+        # self.authorizers = authorizers or dict()
+        # self.auto_login = auto_login
         self.auto_registration = auto_registration
 
-        private_cfg = self.get_cfg(private=True)
-        if gladier.utils.config_migrations.needs_migration(private_cfg):
-            private_cfg = gladier.utils.config_migrations.migrate_gladier(private_cfg)
-            private_cfg.save()
-
-        if os.path.exists(self.config_filename):
-            pub_cfg = self.get_cfg(private=False)
-            if gladier.utils.config_migrations.needs_migration(pub_cfg):
-                pub_cfg = gladier.utils.config_migrations.migrate_gladier(pub_cfg)
-                pub_cfg.save()
-
-        if self.authorizers and self.auto_login:
-            log.warning('Authorizers provided when "auto_login=True", you probably want to set '
-                        'auto_login=False if you are providing your own authorizers...')
-        if self.authorizers and self.missing_authorizers:
-            raise gladier.exc.AuthException(f'Missing Authorizers: {self.missing_authorizers}')
-        try:
-            if not self.authorizers:
-                log.debug('No authorizers provided, loading from disk.')
-                self.authorizers = self.get_native_client().get_authorizers_by_scope()
-        except fair_research_login.exc.LoadError:
-            log.debug('Load from disk failed, login will be required.')
-        if self.auto_login and not self.is_logged_in():
-            self.login()
-
-    def _load_public_config(self):
-        return gladier.config.GladierConfig(self.config_filename, self.section)
-
-    def _load_private_config(self):
-        return gladier.config.GladierSecretsConfig(self.secret_config_filename,
-                                                   self.section, self.client_id)
+        if storage is None:
+            class_name = self.__class__.__name__
+            section = gladier.utils.name_generation.get_snake_case(class_name)
+            storage = gladier.config.GladierSecretsConfig(
+                self.secret_config_filename, section, self.client_id
+            )
+            storage.update()
+        self.storage = storage
+        self.login_manager = login_manager or LoginManager(self.client_id, self.storage,
+                                                           self.app_name)
+        self.flows_manager = flows_manager or FlowsManager(login_manager=self.login_manager,
+                                                           storage=self.storage)
+        self.funcx_manager = funcx_manager or FuncXManager(login_manager=self.login_manager,
+                                                           storage=self.storage)
```

Comment:
This is probably the most interesting change to the Gladier Client class. Separating out the different managers also allows for customizing them to different needs. This will be vital for portals and other native apps which want to build on the Gladier SDK and need to use custom login/storage solutions. 

We could also allow modifying the flows/funcx manager classes, but I'm tempted to take those off the list since I'm not sure there's much in there we want to encourage folks to customize (Customizing funcx function registration is out of scope here).